lib/matplotlib/text.py

```diff
@@ -165,6 +165,37 @@ def __init__(self,
         self._x, self._y = x, y
         self._text = ''
         self.set_text(text)
```

Comment:
Sorry, just 
```
self._text = ''
```
should be here (to initialize the `_text` argument).

`self.set_text(text)` should still go in `_reset_visual_defaults`. If not, it will not clear the text contents. (If it works, it is a sign of something else being dodgy.)